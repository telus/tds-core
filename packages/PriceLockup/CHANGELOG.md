# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [2.1.1](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.1.0...@tds/core-price-lockup@2.1.1) (2020-07-22)

**Note:** Version bump only for package @tds/core-price-lockup





# [2.1.0](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.28...@tds/core-price-lockup@2.1.0) (2020-04-14)


### Features

* **core-price-lockup:** add footnoteLinks prop ([cd0d1bf](https://github.com/telusdigital/tds-core/commit/cd0d1bfecea6bcd308c3e06ac7387fcc8f0017a7))





## [2.0.28](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.27...@tds/core-price-lockup@2.0.28) (2020-03-11)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.27](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.26...@tds/core-price-lockup@2.0.27) (2020-03-10)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.26](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.25...@tds/core-price-lockup@2.0.26) (2020-03-09)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.25](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.24...@tds/core-price-lockup@2.0.25) (2020-02-24)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.24](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.23...@tds/core-price-lockup@2.0.24) (2020-02-03)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.23](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.22...@tds/core-price-lockup@2.0.23) (2020-01-27)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.22](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.21...@tds/core-price-lockup@2.0.22) (2020-01-23)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.21](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.20...@tds/core-price-lockup@2.0.21) (2020-01-22)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.20](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.19...@tds/core-price-lockup@2.0.20) (2019-12-18)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.19](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.18...@tds/core-price-lockup@2.0.19) (2019-11-18)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.18](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.17...@tds/core-price-lockup@2.0.18) (2019-10-09)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.17](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.16...@tds/core-price-lockup@2.0.17) (2019-10-04)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.16](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.15...@tds/core-price-lockup@2.0.16) (2019-09-12)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.15](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.14...@tds/core-price-lockup@2.0.15) (2019-09-06)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.14](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.13...@tds/core-price-lockup@2.0.14) (2019-08-22)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.13](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.12...@tds/core-price-lockup@2.0.13) (2019-08-09)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.12](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.11...@tds/core-price-lockup@2.0.12) (2019-08-01)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.11](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.10...@tds/core-price-lockup@2.0.11) (2019-07-30)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.10](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.9...@tds/core-price-lockup@2.0.10) (2019-07-25)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.9](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.8...@tds/core-price-lockup@2.0.9) (2019-07-17)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.8](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.7...@tds/core-price-lockup@2.0.8) (2019-07-10)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.7](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.6...@tds/core-price-lockup@2.0.7) (2019-06-25)

**Note:** Version bump only for package @tds/core-price-lockup





## [2.0.6](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.5...@tds/core-price-lockup@2.0.6) (2019-06-04)

**Note:** Version bump only for package @tds/core-price-lockup

## [2.0.5](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.4...@tds/core-price-lockup@2.0.5) (2019-05-22)

**Note:** Version bump only for package @tds/core-price-lockup

## [2.0.4](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.3...@tds/core-price-lockup@2.0.4) (2019-05-07)

**Note:** Version bump only for package @tds/core-price-lockup

## [2.0.3](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.2...@tds/core-price-lockup@2.0.3) (2019-05-01)

**Note:** Version bump only for package @tds/core-price-lockup

## [2.0.2](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.1...@tds/core-price-lockup@2.0.2) (2019-05-01)

**Note:** Version bump only for package @tds/core-price-lockup

## [2.0.1](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@2.0.0...@tds/core-price-lockup@2.0.1) (2019-04-23)

**Note:** Version bump only for package @tds/core-price-lockup

# [2.0.0](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.16...@tds/core-price-lockup@2.0.0) (2019-04-22)

### Features

- **core-price-lockup:** upgrade to styled-components ([f42e0b4](https://github.com/telusdigital/tds-core/commit/f42e0b4))

### BREAKING CHANGES

- **core-price-lockup:** upgrade to styled-components

## [1.0.16](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.15...@tds/core-price-lockup@1.0.16) (2019-04-11)

### Bug Fixes

- **core-price-lockup:** use className prop for Box ([2fba1cb](https://github.com/telusdigital/tds-core/commit/2fba1cb))

## [1.0.15](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.14...@tds/core-price-lockup@1.0.15) (2019-04-08)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.14](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.13...@tds/core-price-lockup@1.0.14) (2019-04-08)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.13](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.12...@tds/core-price-lockup@1.0.13) (2019-03-28)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.12](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.11...@tds/core-price-lockup@1.0.12) (2019-03-28)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.11](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.10...@tds/core-price-lockup@1.0.11) (2019-03-20)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.10](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.9...@tds/core-price-lockup@1.0.10) (2019-03-20)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.9](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.8...@tds/core-price-lockup@1.0.9) (2019-03-18)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.8](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.7...@tds/core-price-lockup@1.0.8) (2019-03-18)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.7](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.6...@tds/core-price-lockup@1.0.7) (2019-03-11)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.6](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.5...@tds/core-price-lockup@1.0.6) (2019-03-05)

**Note:** Version bump only for package @tds/core-price-lockup

## [1.0.5](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.4...@tds/core-price-lockup@1.0.5) (2019-02-06)

**Note:** Version bump only for package @tds/core-price-lockup

<a name="1.0.4"></a>

## [1.0.4](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.3...@tds/core-price-lockup@1.0.4) (2018-12-19)

### Bug Fixes

- **core-price-lockup:** prevent improper line height ([ae5117d](https://github.com/telusdigital/tds-core/commit/ae5117d))
- **core-price-lockup:** use shared typography stylesheet ([8b092df](https://github.com/telusdigital/tds-core/commit/8b092df))

<a name="1.0.3"></a>

## [1.0.3](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.2...@tds/core-price-lockup@1.0.3) (2018-12-11)

### Bug Fixes

- **core-price-lockup:** fixing horizontal alignment issue ([#836](https://github.com/telusdigital/tds-core/issues/836)) ([5a64f4e](https://github.com/telusdigital/tds-core/commit/5a64f4e))

<a name="1.0.2"></a>

## [1.0.2](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.1...@tds/core-price-lockup@1.0.2) (2018-12-06)

**Note:** Version bump only for package @tds/core-price-lockup

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telusdigital/tds-core/compare/@tds/core-price-lockup@1.0.0...@tds/core-price-lockup@1.0.1) (2018-11-26)

### Bug Fixes

- **core-price-lockup:** large lockup \$ position ([605b600](https://github.com/telusdigital/tds-core/commit/605b600))
- **core-price-lockup:** reduce build size ([78d35c5](https://github.com/telusdigital/tds-core/commit/78d35c5))

<a name="1.0.0"></a>

# 1.0.0 (2018-11-13)

### Bug Fixes

- **core-price-lockup:** add core-text to devDependencies ([a1a65a8](https://github.com/telusdigital/tds-core/commit/a1a65a8))
- **core-price-lockup:** add defaultprops ([48273f8](https://github.com/telusdigital/tds-core/commit/48273f8))
- **core-price-lockup:** adding priceBold Prop and style ([066abff](https://github.com/telusdigital/tds-core/commit/066abff))
- **core-price-lockup:** adjustments based comments ([63400c9](https://github.com/telusdigital/tds-core/commit/63400c9))
- **core-price-lockup:** align rateText to bottom with size small ([3eb27a6](https://github.com/telusdigital/tds-core/commit/3eb27a6))
- **core-price-lockup:** dollar sign should be md when lockup is sm ([f16264a](https://github.com/telusdigital/tds-core/commit/f16264a))
- **core-price-lockup:** fix lint issue in scss pricelockup ([510de89](https://github.com/telusdigital/tds-core/commit/510de89))
- **core-price-lockup:** install dependencies ([86568eb](https://github.com/telusdigital/tds-core/commit/86568eb))
- **core-price-lockup:** lint issue again, added newline ([0114125](https://github.com/telusdigital/tds-core/commit/0114125))
- **core-price-lockup:** new unit tests, ui spacing corrections ([b67aa13](https://github.com/telusdigital/tds-core/commit/b67aa13))
- **core-price-lockup:** remove composition of text styles ([6fe6833](https://github.com/telusdigital/tds-core/commit/6fe6833))
- **core-price-lockup:** remove core-colours/colours ([b1cf5a2](https://github.com/telusdigital/tds-core/commit/b1cf5a2))
- **core-price-lockup:** remove warnings lint ([2497216](https://github.com/telusdigital/tds-core/commit/2497216))
- **core-price-lockup:** spacing between topText, price, bottomText ([a99752d](https://github.com/telusdigital/tds-core/commit/a99752d))
- **core-price-lockup:** tests and streamline code ([cf803a4](https://github.com/telusdigital/tds-core/commit/cf803a4))
- **core-price-lockup:** update spacings ([ed82d91](https://github.com/telusdigital/tds-core/commit/ed82d91))

### Features

- **core-price-lockup:** building and contributing tds-core-price-lock ([dbd0c03](https://github.com/telusdigital/tds-core/commit/dbd0c03))
- **core-price-lockup:** new tests and render method ([4cc99ca](https://github.com/telusdigital/tds-core/commit/4cc99ca))
- **core-price-lockup:** updated readme ([9b2e4ad](https://github.com/telusdigital/tds-core/commit/9b2e4ad))
