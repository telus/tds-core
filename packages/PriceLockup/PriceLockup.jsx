import React from 'react'
import PropTypes from 'prop-types'
import Heading from '@tds/core-heading'
import Text from '@tds/core-text'
import DisplayHeading from '@tds/core-display-heading'
import HairlineDivider from '@tds/core-hairline-divider'
import styles from './PriceLockup.modules.scss'
import safeRest from '../../shared/utils/safeRest'

/**
 * A component presenting Telus product pricing information.
 * @version ./package.json
 */

const TopTextStyleDecider = componentSize => {
  switch (componentSize) {
    case 'small':
    case 'medium':
      return 'small'
    case 'large':
      return 'large'
    default:
      return ''
  }
}

const DollarSignStyleDecider = componentSize => {
  switch (componentSize) {
    case 'small':
      return 'medium'
    case 'medium':
      return 'large'
    default:
      return ''
  }
}

const RateTextSizeDecider = componentSize => {
  switch (componentSize) {
    case 'small':
    case 'medium':
      return 'medium'
    case 'large':
      return 'large'
    default:
      return ''
  }
}

const PriceLockup = ({ size, price, topText, signDirection, rateText, bottomText, ...rest }) => {
  const TopText = () => {
    const topTextSize = TopTextStyleDecider(size)
    return <Text size={topTextSize}>{topText}</Text>
  }

  const BottomText = () => {
    return <Text size={size}>{bottomText}</Text>
  }

  const DollarSign = () => {
    if (size === 'large') {
      return <h1 className={styles.dollarSignH1Style}>&#36;</h1>
    }
    const DollarSignSize = DollarSignStyleDecider(size)
    return <Text size={DollarSignSize}>&#36;</Text>
  }

  const DollarValue = () => {
    if (size === 'small' || 'medium') {
      return <Heading level={size === 'small' ? 'h2' : 'h1'}>{price}</Heading>
    }
    return <DisplayHeading>{price}</DisplayHeading>
  }

  const RateText = () => {
    const RateTextSize = RateTextSizeDecider(size)
    return (
      <div className={size === 'small' ? styles.rateWrapperSmall : styles.rateWrapperMedium}>
        <Text size={RateTextSize}>{rateText}</Text>
      </div>
    )
  }

  const PriceValueSign = () => {
    return (
      <span className={styles.priceValueSign}>
        {signDirection === 'left' ? <DollarSign /> : undefined}
        <DollarValue />
        {signDirection === 'right' ? <DollarSign /> : undefined}
      </span>
    )
  }

  const Hairline = () => {
    if (rateText && bottomText && size === 'medium') {
      return (
        <div className={styles.hairlineWrapper}>
          <HairlineDivider />
        </div>
      )
    }
    return ''
  }

  return (
    <div {...safeRest(rest)}>
      <TopText />
      <div className={styles.priceWrapper}>
        <PriceValueSign />
        <RateText />
      </div>
      <Hairline />
      {size === 'medium' ? <BottomText /> : ''}
    </div>
  )
}

PriceLockup.propTypes = {
  /**
   * Font Size of Labels, Price, and Signs
   */
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  /**
   * Position of Dollar Sign relative to Price Value
   */
  signDirection: PropTypes.oneOf(['left', 'right']),
  /**
   * Statement above Price Value
   */
  topText: PropTypes.string,
  /**
   * Statement below Price Value
   */
  bottomText: PropTypes.string,
  /**
   * Statement right of Price Value
   */
  rateText: PropTypes.string,
  /**
   * Price value of component
   */
  price: PropTypes.string.isRequired,
}

PriceLockup.defaultProps = {
  size: undefined,
  signDirection: undefined,
  topText: undefined,
  bottomText: undefined,
  rateText: undefined,
}

export default PriceLockup
