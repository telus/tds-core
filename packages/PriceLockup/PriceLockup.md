```jsx
<PriceLockup
  size="medium"
  topText="Starting at"
  bottomText="$68 /month after 3 months"
  rateText="/month"
  price="25"
  signDirection="left"
/>
```
