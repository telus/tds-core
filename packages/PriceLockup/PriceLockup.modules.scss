@import '~@tds/core-colours/colours'; // be sure to add core-colours as a dependency first

.hairlineWrapper {
  padding: 0.5rem;
}

.priceValueSign {
  display: flex;
  composes: h1 from '../Heading/Heading.modules.scss';
  h1, h2 {
    color: $color-text;
  }
}

.priceWrapper {
  display: flex;
  align-items: flex-end;
}

.rateWrapperSmall {
  padding-bottom: 0.1rem;
}

.rateWrapperMedium {
  padding-bottom: 0.2rem;
}

.dollarSignH1Style {
  font-size: 1.8rem;
  padding-top: 0.3rem;
}
