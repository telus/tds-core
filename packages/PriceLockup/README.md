# TDS Core: PriceLockup

Element component that can display product price, rate, top and bottom text if necessary, along with a dollar sign that can be adjusted left and right.
