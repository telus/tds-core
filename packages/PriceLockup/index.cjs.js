require('./dist/index.css')
const PriceLockup = require('./dist/index.cjs')

module.exports = PriceLockup
