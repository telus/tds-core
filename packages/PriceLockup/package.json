{
  "name": "@tds/core-price-lockup",
  "version": "1.0.0-preview",
  "description": "",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds-core.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds-core/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  },
  "dependencies": {
    "prop-types": "^15.5.10"
  }
}
