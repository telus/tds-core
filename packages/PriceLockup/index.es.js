import './dist/index.css'
import PriceLockup from './dist/index.es'

export default PriceLockup
