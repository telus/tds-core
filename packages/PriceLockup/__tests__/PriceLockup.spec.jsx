import React from 'react'
import { shallow } from 'enzyme'

import PriceLockup from '../PriceLockup'

describe('PriceLockup', () => {
  const defaultProps = {
    price: '25',
  }
  const doShallow = (overrides = {}) => shallow(<PriceLockup {...defaultProps} {...overrides} />)
  it('renders', () => {
    const priceLockup = doShallow()

    expect(priceLockup).toMatchSnapshot()
  })

  it('passes additional attributes to the element', () => {
    const priceLockup = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(priceLockup).toHaveProp('id', 'the-id')
    expect(priceLockup).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const priceLockup = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(priceLockup).not.toHaveProp('className', 'my-custom-class')
    expect(priceLockup).not.toHaveProp('style')
  })

  it('will render BottomText when size prop is medium', () => {
    const priceLockup = doShallow({ size: 'medium' })
    expect(priceLockup.text()).toContain('<BottomText />')
  })

  it('will not render BottomText when size prop is not medium', () => {
    const priceLockup = doShallow({ size: 'large' })
    expect(priceLockup.text()).not.toContain('<BottomText />')
  })
})
