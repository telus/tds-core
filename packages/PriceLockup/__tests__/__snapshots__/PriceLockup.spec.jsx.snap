// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceLockup renders 1`] = `
<div>
  <TopText />
  <div
    className="priceWrapper"
  >
    <PriceValueSign />
    <RateText />
  </div>
  <Hairline />
</div>
`;
